python/lib/dependabot/python/file_fetcher.rb

```diff
@@ -337,10 +337,7 @@ def fetch_path_setup_file(path, allow_pyproject: false)
             ).tap { |f| f.support_file = true }
           rescue Dependabot::DependencyFileNotFound
             # For projects with pyproject.toml attempt to fetch a pyproject.toml
-            # at the given path instead of a setup.py. We do not require a
-            # setup.py to be present, so if none can be found, simply return
-            return [] unless allow_pyproject
-
+            # at the given path instead of a setup.py.
```

Comment:
Won't this cause python projects that don't use `pyproject.toml` to raise an error when they call `#fetch_path_setup_file`?

Do all python ecosystems that we support now use `pyproject.toml`?

Reply:
If there's a "path dependency" (what Python calls "editable" dependency), then there needs to be either a setup.py or a pyproject.toml file at that path. If none of them are found, at pip-compile does not consider it a valid Python project and fails I believe.

In this case, we're in the rescue block from fetching a setup.py file, so this means none of them are found.

Reply:
That makes sense

> If there's a "path dependency" (what Python calls "editable" dependency), then there needs to be either a setup.py or a pyproject.toml file at that path

Does your code support when a pyproject.toml is found but a setup.py is not?

I see your test at https://github.com/dependabot/dependabot-core/pull/8172/files#diff-ba69c800c882181677c61aa2162b09edfd1bb30422d39e5f11afe35dcf2df821R833-R846 only checks whether a `DependencyFileNotFound` error is raised when neither exist

Reply:
Ah I misread the path.gsub 

Thanks for explaining this all to me!